modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/extension/discovery/RepositoryLookupExtensionModelGenerator.java

```diff
@@ -73,23 +77,24 @@ public ExtensionModel obtainExtensionModel(ExtensionDiscoveryRequest discoveryRe
    * @param artifactName                   the name of the artifact being loaded.
    * @param additionalAttributes           custom parameters for the
    *                                       {@link org.mule.runtime.extension.api.loader.ExtensionModelLoadingRequest}.
+   * @param artifactPluginDescriptor
    * @throws IllegalArgumentException there is no {@link ExtensionModelLoader} for the ID in the {@link MulePluginModel}.
    */
   private ExtensionModel discoverExtensionThroughJsonDescriber(ExtensionModelLoaderRepository extensionModelLoaderRepository,
                                                                LoaderDescriber loaderDescriber,
                                                                Set<ExtensionModel> dependencies,
                                                                Supplier<ClassLoader> artifactClassloader,
                                                                String artifactName,
-                                                               Map<String, Object> additionalAttributes) {
+                                                               Map<String, Object> additionalAttributes,
+                                                               ArtifactPluginDescriptor artifactPluginDescriptor) {
     ExtensionModelLoader loader = extensionModelLoaderRepository.getExtensionModelLoader(loaderDescriber)
         .orElseThrow(() -> new IllegalArgumentException(format("The identifier '%s' does not match with the describers available "
             + "to generate an ExtensionModel (working with the plugin '%s')", loaderDescriber.getId(), artifactName)));
     Map<String, Object> attributes = new HashMap<>(loaderDescriber.getAttributes());
     attributes.putAll(additionalAttributes);
 
     return loader.loadExtensionModel(builder(artifactClassloader.get(), getDefault(dependencies))
-        .addParameters(attributes)
+        .addParameters(attributes).addEnricher(new ArtifactCoordinatesEnricher(artifactPluginDescriptor.getBundleDescriptor()))
```

Comment:
shouldn't the enricher be added in the same place where the rest of the enrichers are declared? @marianogonzalez

Reply:
it depends. If this is something that must only be added when the EM is created through this code path, then it belongs here.

If it's something we want enforced at a broader scope then it should be added at either the loader or ExtensionModelFactory levels. It depends on the use case.